test/functional/editions_controller_test.rb

```diff
@@ -520,84 +520,167 @@ class EditionsControllerTest < ActionController::TestCase
-        @edition.reload
-        assert_equal "scheduled_for_publishing", @edition.state
+          assert_equal "You do not have correct editor permissions for this action.", flash[:danger]
+        end
       end
 
-      should "notify the publishing API to publish the edition" do
-        PublishWorker.expects(:perform_async).with(@edition.id.to_s)
+      context "edition is not in a valid state to be published" do
+        setup do
+          @requester = FactoryBot.create(:user, :govuk_editor, name: "Stub Requester")
+          @edition = FactoryBot.create(:edition, state: "draft")
+        end
 
-        post :send_to_publish, params: {
-          id: @edition.id,
-        }
+        should "not update the edition state and render 'send_to_publish' template with an error" do
+          post :send_to_publish, params: {
+            id: @edition.id,
+          }
+
+          assert_equal "Edition is not in a state where it can be published", flash[:danger]
+
+          assert_template "secondary_nav_tabs/send_to_publish_page"
+          @edition.reload
+          assert_equal "draft", @edition.state
+        end
       end
     end
 
-    context "user does not have govuk_editor permission" do
+    context "edition is in 'ready' state" do
       setup do
-        user = FactoryBot.create(:user)
-        login_as(user)
+        @edition = FactoryBot.create(:edition, state: "ready")
+        login_as_stub_user
       end
 
-      should "render an error message" do
-        post :send_to_publish, params: {
-          id: @edition.id,
-        }
+      context "user has govuk_editor permission" do
+        setup do
+          PublishWorker.stubs(:perform_async)
+        end
 
-        assert_equal "You do not have correct editor permissions for this action.", flash[:danger]
+        should "update the edition status to 'published' and save the comment" do
+          post :send_to_publish, params: {
+            id: @edition.id,
+            comment: "Publishing this ready edition",
+          }
+
+          assert_equal "Published", flash[:success]
+          @edition.reload
+          assert_equal "published", @edition.state
+          assert_equal "Publishing this ready edition", @edition.latest_status_action.comment
+          assert_equal @user.id, @edition.latest_status_action.requester_id
+        end
+
+        should "not update the edition state and render 'send_to_publish' template with an error when an error occurs" do
+          EditionProgressor.any_instance.expects(:progress).returns(false)
+
+          post :send_to_publish, params: {
+            id: @edition.id,
+          }
+
+          assert_template "secondary_nav_tabs/send_to_publish_page"
+          assert_equal "Due to a service problem, the request could not be made", flash[:danger]
+          @edition.reload
+          assert_equal "ready", @edition.state
+        end
+
+        should "notify the publishing API to publish the edition" do
```

Comment:
As above, this has no assertions. Is that correct?

Reply:
Indeed, that is correct, no need, as it is tested via a stub, so incorrect string values not being the edition id would cause it to fail.